### DISCUSSION TITLE
HASPmota example

### DISCUSSION THREAD

**Discussion by wapoone** [arendst/Tasmota#16297] [Q&A] [answered]
Hello, i read through the HASPmota wiki, but i still dont understand a thing.
how do i toggle on a circle when POWER1 is enabled/ toggled off when POWER1 is disabled.
The code looks like dis at the moment:
`{"id":15,"obj":"spinner","x":327,"y":150,"w":30,"h":30,"bg_opa":0,"border_width":0,"line_width":5,"line_width10":5,"type":2,"angle":120,"speed":1000}`
Thanks in advance

**Comment by s-hadinger** [comment]
The `spinner` type is based on the LVGL `spinner` object. Unfortunately the LVGL implementation doesn't allow to change the spinner once it's created. Internally the spinner is actually an `arc` with an animation attached.

There should be a way to get back the animation object of the spinner and change its properties. I'll need to dig into this.

**Reply by wapoone** [reply]
I can use hidden on or off . 
`"hidden":0` or `"hidden":1`
how do i get the status of POWER1 ?

**Reply by s-hadinger** [reply]
You can do something like:

```
tasmota.add_rule("Power1#State=1", def () global.p1b15.hidden = true end)
tasmota.add_rule("Power1#State=0", def () global.p1b15.hidden = false end)
```

**Reply by sfromis** [reply]
That could be compressed a bit to
`tasmota.add_rule("Power1#State", def (state) global.p1b15.hidden = state==1 end)`
:grinning:

**Reply by s-hadinger** [reply]
Well done

**Reply by wapoone** [reply]
Thanks. Where is the best place to enter this? My attempts have all failed 😀

**Comment by s-hadinger** [comment, marked as answer]
See https://tasmota.github.io/docs/HASPmota/#run-arbitrary-berry-code

You can add this line to `pages.jsonl`

Note: I inferred that it was on page 1 hence p1b15

`{"berry_run":"tasmota.add_rule('Power1#State', def (state) global.p1b15.hidden = state==1 end)"}`

**Reply by wapoone** [reply]
Thank you very much. 
I've already tried that with no result. I changed my code like this. 

`{"id":15,"obj":"spinner","x":327,"y":150,"w":30,"h":30,"bg_opa":0,"border_width":0,"line_width":5,"line_width10":5,"type":2,"angle":120,"speed":1000,"hidden":"POWER1#State"}`

now it works, albeit in reverse

POWER1 ON - Spinner invisible, POWER1 OFF - Spinner visible
can i change this?

**Reply by s-hadinger** [reply]
I'm sorry but this can't work. `"hidden":"POWER1#State"` is not valid

**Reply by wapoone** [reply]
I probably misunderstood something or wrote something wrong. 
my code is like this at the moment
`{"berry_run":"tasmota.add_rule('POWER1#State', def (state) global.p1b15.hidden = state==1 end)"}
{"id":15,"obj":"spinner","x":327,"y":150,"w":30,"h":30,"bg_opa":0,"border_width":0,"line_width":5,"line_width10":5,"type":2,"angle":120,"speed":1000,"hidden":"POWER1#State"}`
and it works. 
How is that right ?
for info have one WT32-SC01

**Reply by sfromis** [reply]
This rule matches what s-hadinger posted as example:
`{"berry_run":"tasmota.add_rule('POWER1#State', def (state) global.p1b15.hidden = state==1 end)"}`
If you want to have it hidden when off, flip the condition:
`{"berry_run":"tasmota.add_rule('POWER1#State', def (state) global.p1b15.hidden = state==0 end)"}`
The point was that the part about using `"hidden":"POWER1#State"` in the spinner object definition cannot work.

**Reply by wapoone** [reply]
Thank you very much for your help. 🙏👍

**Comment by wapoone** [comment]
need the help again.
Can't get along with examples in the wiki.
when i touch an object i want to switch a relay.
for example click there
`{"id":15,"obj":"lv_wifi_arcs","x":430,"y":0,"w":29,"h":22,"radius":0,"border_side":0,"bg_color":"#000000","line_color":"#FFFFFF"}`
and `POWER2` switching
 What does the rule look like?
Thanks
